File: app/src/main/java/com/datarockets/mnchkn/activities/onboard/OnboardPresenterImpl.java
package com.datarockets.mnchkn.activities.onboard;

public class OnboardPresenterImpl implements OnboardPresenter,
        OnboardInteractorImpl.OnFinishedChecking {

    private OnboardView mOnboardView;
    private OnboardInteractor mOnboardInteractor;

    public OnboardPresenterImpl(OnboardView onboardView, OnboardInteractor onboardInteractor) {
        this.mOnboardView = onboardView;
        this.mOnboardInteractor = onboardInteractor;
    }

    @Override
    public void checkIsUserSeenOnboarding() {
        if (mOnboardView != null) {
            mOnboardInteractor.isUserSeenOnboarding(this);
        }
    }

    @Override
    public void setOnboardingSeen() {
        if (mOnboardView != null) {
            mOnboardInteractor.setOnboardingSeen();
        }
    }

    @Override
    public void onDestroy() {
        if (mOnboardView != null) {
            mOnboardView = null;
        }
    }

    @Override
    public void shouldShowOnboarding(boolean value) {
        if (mOnboardView != null && !value) {
            mOnboardView.openPlayersActivity();
        }
    }
}

File: app/src/main/kotlin/com/datarockets/mnchkn/ui/onboard/OnboardPresenterImpl.kt
package com.datarockets.mnchkn.ui.onboard

class OnboardPresenterImpl(private var mOnboardView: OnboardView?,
                           private val mOnboardInteractor: OnboardInteractor)
: OnboardPresenter, OnboardInteractor.OnFinishedChecking {

    override fun checkIsUserSeenOnboarding() {
        if (mOnboardView != null) {
            mOnboardInteractor.isUserSeenOnboarding(this)
        }
    }

    override fun setOnboardingSeen() {
        if (mOnboardView != null) {
            mOnboardInteractor.setOnboardingSeen()
        }
    }

    override fun onDestroy() {
        if (mOnboardView != null) {
            mOnboardView = null
        }
    }

    override fun shouldShowOnboarding(value: Boolean) {
        if (!value) {
            mOnboardView?.openPlayersActivity()
        }
    }
}
